src/runtime-rs/crates/service/src/task_service.rs

```diff
@@ -0,0 +1,184 @@
+use ttrpc::{self, r#async::TtrpcContext};
+
+use runtimes::RuntimeHandlerManager;
+
+pub(crate) struct TaskService {
+    handler: Arc<RuntimeHandlerManager>,
+}
+
+impl TaskService {
+    pub(crate) fn new(handler: Arc<RuntimeHandlerManager>) -> Self {
+        Self { handler }
+    }
+}
+
+async fn handler_message<TtrpcReq, TtrpcResp>(
+    s: &RuntimeHandlerManager,
+    ctx: &TtrpcContext,
+    req: TtrpcReq,
+) -> ttrpc::Result<TtrpcResp>
+where
+    Request: TryFrom<TtrpcReq>,
+    <Request as TryFrom<TtrpcReq>>::Error: std::fmt::Debug,
+    TtrpcResp: TryFrom<Response>,
+    <TtrpcResp as TryFrom<Response>>::Error: std::fmt::Debug,
+{
+    let r = req
+        .try_into()
+        .map_err(|err| ttrpc::Error::Others(format!("failed to translate from shim {:?}", err)))?;
+    let logger = sl!().new(o!("steam id" =>  ctx.mh.stream_id));
+    debug!(logger, "====> task service {:?}", &r);
+    let resp = s
+        .handler_message(r)
+        .await
+        .map_err(|err| ttrpc::Error::Others(format!("failed to handler message {:?}", err)))?;
+    debug!(logger, "<==== task service {:?}", &resp);
+    Ok(resp
+        .try_into()
+        .map_err(|err| ttrpc::Error::Others(format!("failed to translate to shim {:?}", err)))?)
+}
+
+#[async_trait]
+impl shim_async::Task for TaskService {
+    async fn state(
+        &self,
+        ctx: &TtrpcContext,
+        req: api::StateRequest,
+    ) -> ttrpc::Result<api::StateResponse> {
+        handler_message(&self.handler, ctx, req).await
+    }
+
+    async fn create(
+        &self,
+        ctx: &TtrpcContext,
+        req: api::CreateTaskRequest,
+    ) -> ttrpc::Result<api::CreateTaskResponse> {
+        handler_message(&self.handler, ctx, req).await
+    }
+
+    async fn start(
+        &self,
+        ctx: &TtrpcContext,
+        req: api::StartRequest,
+    ) -> ttrpc::Result<api::StartResponse> {
+        handler_message(&self.handler, ctx, req).await
+    }
+
+    async fn delete(
+        &self,
+        ctx: &TtrpcContext,
+        req: api::DeleteRequest,
+    ) -> ttrpc::Result<api::DeleteResponse> {
+        handler_message(&self.handler, ctx, req).await
+    }
+
+    async fn pids(
+        &self,
+        ctx: &TtrpcContext,
+        req: api::PidsRequest,
+    ) -> ttrpc::Result<api::PidsResponse> {
+        handler_message(&self.handler, ctx, req).await
+    }
+
+    async fn pause(&self, ctx: &TtrpcContext, req: api::PauseRequest) -> ttrpc::Result<api::Empty> {
+        handler_message(&self.handler, ctx, req).await
+    }
+
+    async fn resume(
+        &self,
+        ctx: &TtrpcContext,
+        req: api::ResumeRequest,
+    ) -> ttrpc::Result<api::Empty> {
+        handler_message(&self.handler, ctx, req).await
+    }
+
+    async fn checkpoint(
+        &self,
+        _ctx: &TtrpcContext,
+        _req: api::CheckpointTaskRequest,
+    ) -> ttrpc::Result<api::Empty> {
+        Err(ttrpc::Error::Others("Not support".to_string()))
```

Comment:
`Not Implemented` may be a standard error message.

Reply:
Thanks for your review, has implemented the function.